x-pack/plugins/cases/common/schema/index.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as rt from 'io-ts';
+import { either } from 'fp-ts/lib/Either';
+
+export const NonEmptyString = new rt.Type<string, string, unknown>(
+  'NonEmptyString',
+  rt.string.is,
+  (input, context) =>
+    either.chain(rt.string.validate(input, context), (s) => {
+      if (s.trim() !== '') {
+        return rt.success(s);
+      } else {
+        return rt.failure(input, context, 'string must have length >= 1');
+      }
+    }),
+  rt.identity
+);
+
+export const limitedArraySchema = <T extends rt.Mixed>(codec: T, min: number, max: number) =>
```

Comment:
this looks like an incredibly complicated way of checking an array size 😆

Reply:
Seems like magic.